anomali-limo-feed/operations.py

```diff
@@ -0,0 +1,217 @@
+                "filename": "",
+                "content_length": "",
+                "content_type": ""
+            })
+        else:
+            return ({
+                "id": "",
+                "objects": [
+                    {
+                        "created": "",
+                        "description": "",
+                        "id": "",
+                        "labels": [
+                        ],
+                        "modified": "",
+                        "name": "",
+                        "object_marking_refs": [
+                        ],
+                        "pattern": "",
+                        "type": "",
+                        "valid_from": ""
+                    }
+                ],
+                "spec_version": "",
+                "type": ""
+            })
+    except Exception as e:
+        logger.exception('An exception occurred {}'.format(e))
+        raise ConnectorError('An exception occurred {}'.format(e))
+
+
+def get_api_root_information(config, params):
+    try:
+        taxii = TaxiiClient(config)
+        params = get_params(params)
+        return taxii.make_request_taxii(params=params)
+    except Exception as e:
+        logger.exception('An exception occurred {}'.format(e))
+        raise ConnectorError('An exception occurred {}'.format(e))
+
+
+def get_collections(config, params):
+    try:
+        taxii = TaxiiClient(config)
+        params = get_params(params)
+        if params:
+            response = taxii.make_request_taxii(endpoint='collections/' + params['collectionID'] + '/')
+        else:
+            response = taxii.make_request_taxii(endpoint='collections/')
+        if response.get('collections'):
+            return response
+        else:
+            return {'collections': [response]}
+    except Exception as e:
+        logger.exception('An exception occurred {}'.format(e))
+        raise ConnectorError('An exception occurred {}'.format(e))
+
+
+def get_objects_by_collection_id(config, params):
+    try:
+        taxii = TaxiiClient(config)
+        if params.get('limit') is None or params.get('limit') == '':
+            server_url = config.get('server_url')
+            if not server_url.startswith('https://'):
+                server_url = 'https://' + server_url
+            if not server_url.endswith('/'):
+                server_url += '/'
+            username = config.get('username')
+            password = config.get('password')
+            collection = Collection(
+                server_url + 'api/v1/taxii2/feeds/collections/' + str(params.get('collectionID')) + '/',
+                user=username, password=password)
+            response = []
+            for bundle in as_pages(collection.get_objects, added_after=params.get('added_after') if (
+                    params.get('added_after') is not None and params.get(
+                'added_after') != '') else '1970-01-01T00:00:00.000Z', start=params.get('offset'),
```

Comment:
what if offset is None? will this fail?

Reply:
if we don't specify an added_after to the function, does it default to today's date or 1970?

Reply:
Offset is a required parameter and limit is not required parameter as we have discussed.
if we don't specify the added_after to the function then it will take 1970